.circleci/config.yml

```diff
@@ -114,6 +114,15 @@ jobs:
        - run: rm -rf /home/circleci/project/.git # CircleCI git caching is likely broken
        - checkout
        - run: ci/do_circle_ci.sh check_format
+   spelling:
```

Comment:
I'm fine either way. I am guessing I'll sometimes want a way to be able to interactively run one and not the other while iterating on a PR at various times -- e.g. while adding and testing new checks in check_format.py, but having them in one CI phase seems OK.

Matt+Harvey I'm curious as you've mentioned this a few times in this PR; why is that so much nicer?

Reply:
Mainly because it's logically related to formatting and IMO it's nice to run a single tool to check/fix all formatting, and I think spelling is part of checking formatting. In any case, I don't feel very strongly about it.

One thing I do think we should do though is not waste another CI slot on this to run spelling in parallel since our slots are limited and containers, etc. still take time to spin up. @zyfjeff can you run this script as part of the format CI job? You can just run two scripts in the same container...

Reply:
Agree, let me change